cmake/prep/options.cmake

```diff
@@ -1,3 +1,5 @@
+option(BUILD_WERROR "Enable -Werror flag." ON)
```

Comment:
On by default, but others can disable it if they choose.

Reply:
Personally I would default to off and just force it on for our CI. The risk is basically that somebody packages Sunshine for their distro, then the package no longer builds warning-free after the distro bumps their GCC version some time later.

As upstream, we can monitor for new warnings in new compiler versions and fix or suppress them, but I don't think we want to put that burden on downstream packagers.

Reply:
I updated it. Only code-ql won't have this option enabled, but I don't see that as an issue.